views/default/navigation/menu/elements/required_modules.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/**
+ * Load AMD modules required by the menu item
+ *
+ * @package Elgg.Core
+ * @subpackage Navigation
+ *
+ * @uses $vars['item'] ElggMenuItem
+ */
+
+$item = elgg_extract('item', $vars);
+if (!$item instanceof ElggMenuItem) {
+	return;
+}
+
+$modules = $item->getRequire();
+if (empty($modules)) {
+	return;
+}
+
+foreach ($modules as $module) {
+	elgg_require_js($module);
```

Comment:
This won't work with AJAX, but using inline `<script>` would be an overkill, in case of an entity menu, for example.